File: 백준/Gold/2295. 세 수의 합/세 수의 합.java
import java.io.*;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        int[] nums = new int[N];
        for (int i = 0; i < N; i++) {
            nums[i] = Integer.parseInt(br.readLine());
        }

        Arrays.sort(nums);

        // 두 숫자의 합 배열 만들기
        Set<Integer> nums_sum = new HashSet<>();
        for (int i = 0; i < N; i++) {
            for (int j = i; j < N; j++) {
                nums_sum.add(nums[i] + nums[j]);
            }
        }

        int answer = 0;
        boolean find = false;
        for (int i = N-1; i >= 0; i--) {
            for (int j = 0; j <= i; j++) {
                int tmp = nums[i] - nums[j];
                if (nums_sum.contains(tmp)) {
                    answer = nums[i];
                    find = true;
                    break;
                }
            }
            if (find) {
                break;
            }
        }

        System.out.print(answer);
    }
}

File: 백준/Gold/2295. 세 수의 합/세 수의 합.py
# x + y + z = k -> x + y = k - z

N = int(input())
nums = []
for _ in range(N):
    num = int(input())
    nums.append(num)

nums.sort()
nums_sum = set()
for n1 in nums:
    for n2 in nums:
        nums_sum.add(n1+n2)

def check():
    global answer
    for i in range(N-1, -1, -1):
        for j in range(i+1): # z랑 k랑 같은 수여도 되니깐 i번째 인덱스도 가능. i이상의 인덱스일 경우 값이 음수 일 수 있는데 k는 자연수이니깐 최대 i까지만
            if nums[i] - nums[j] in nums_sum:
                answer = nums[i]
                return

answer = 0
check()
print(answer)